core/conversion/converters/impl/conv_deconv.cpp

```diff
@@ -9,74 +9,147 @@ namespace conversion {
-      auto stride = util::toDims(args[3].unwrapToIntList());
-      LOG_DEBUG("stride: " << stride);
-      auto padding = util::toDims(args[4].unwrapToIntList());
-      LOG_DEBUG("padding: " << padding);
-      auto dilation = util::toDims(args[5].unwrapToIntList());
-      LOG_DEBUG("dilation: " << dilation);
-      bool transposed = args[6].unwrapToBool();
-      auto out_padding = util::toDims(args[7].unwrapToIntList());
-      LOG_DEBUG("out_padding: " << out_padding);
-      int64_t groups = args[8].unwrapToInt();
-
-      nvinfer1::ILayer* new_layer;
-      if (transposed) {
-        nvinfer1::IDeconvolutionLayer* deconv;
-        if (args[2].IValue()->isTensor()) {
-          Weights b(ctx, args[2].IValue()->toTensor());
-          deconv = ctx->net->addDeconvolutionNd(*in, w.num_input_maps, w.kernel_shape, w.data, b.data);
-        } else {
-          deconv = ctx->net->addDeconvolutionNd(*in, w.num_input_maps, w.kernel_shape, w.data, {});
-        }
-
-        TRTORCH_CHECK(deconv, "Unable to create deconvolution layer from node: " << *n);
-
-        deconv->setStrideNd(stride);
-        deconv->setPaddingNd(padding);
+                                 bool deterministic, bool cudnn_enabled, bool allow_tf32) -> (Tensor))SIG",
+            [](ConversionCtx* ctx, const torch::jit::Node* n, args& args) -> bool {
+              auto in = args[0].ITensor(); // assumes non-static input Tensor
+
+              auto w = Weights(ctx, args[1].unwrapToTensor());
+              auto stride = util::toDims(args[3].unwrapToIntList());
+              LOG_DEBUG("stride: " << stride);
+              auto padding = util::toDims(args[4].unwrapToIntList());
+              LOG_DEBUG("padding: " << padding);
+              auto dilation = util::toDims(args[5].unwrapToIntList());
+              LOG_DEBUG("dilation: " << dilation);
+              bool transposed = args[6].unwrapToBool();
+              auto out_padding = util::toDims(args[7].unwrapToIntList());
+              LOG_DEBUG("out_padding: " << out_padding);
+              int64_t groups = args[8].unwrapToInt();
+
+              nvinfer1::ILayer* new_layer;
+              if (transposed) {
+                nvinfer1::IDeconvolutionLayer* deconv;
+                if (args[2].IValue()->isTensor()) {
+                  Weights b(ctx, args[2].IValue()->toTensor());
+                  deconv = ctx->net->addDeconvolutionNd(*in, w.num_input_maps, w.kernel_shape, w.data, b.data);
+                } else {
+                  deconv = ctx->net->addDeconvolutionNd(*in, w.num_input_maps, w.kernel_shape, w.data, {});
+                }
+
+                TRTORCH_CHECK(deconv, "Unable to create deconvolution layer from node: " << *n);
+
+                deconv->setStrideNd(stride);
+                deconv->setPaddingNd(padding);
+#if NV_TENSORRT_MAJOR > 7 || (NV_TENSORRT_MAJOR == 7 && NV_TENSORRT_MINOR == 1)
+                deconv->setDilationNd(dilation);
+                deconv->setNbGroups(groups);
+#endif
```

Comment:
can you add an else which throws a warning if running in pre TRT 7.1 that dilation wont work?

Reply:
Done